spring-integration-core/src/test/java/org/springframework/integration/dsl/routers/RouterTests.java

```diff
@@ -898,6 +917,30 @@ public IntegrationFlow propagateErrorFromGatherer(TaskExecutor taskExecutor) {
 							sg -> sg.gatherTimeout(100))
 					.get();
 		}
+		@Bean
+		public IntegrationFlow scatterGatherflow() {
```

Comment:
So, when you change this flow definition to the simpler one I provided in the comments here in the PR, I'm going to accept it and merge.

Reply:
Hi @artembilan,

I have made the requested changes and pushed the changes to GIT.

I will work on simplifying the scatter gather by removing additional headers in the messages and taking the same from requestMessage context in handleRequestMessage.

This will take some time as I will have to review the impact one more time.